rxnorm/rxnorm_extract.py

```diff
@@ -0,0 +1,181 @@
+    concept_without_father = ['rxnorm', 'others']
+    is_a_dict = {'rxnorm': list(concept_id_dict_key_rxn), 'others': list(concept_id_dict_key_no_rxn)}
+
+    # Saving the relevant data we extracted to json files
+    with open("rxnorm/concepts.json", 'w') as f:
+        json.dump(concept_dict, f)
+
+    with open("rxnorm/is_a.json", 'w') as f:
+        json.dump(is_a_dict, f)
+
+    with open("rxnorm/concept_without_father.json", 'w') as f:
+        json.dump(concept_without_father, f)
+
+    return concept_dict, is_a_dict, concept_without_father
+
+# A function that removes duplicate synonyms between concept alias lists and returns concept dict in the same format it received
+def del_dup_syn(concept_dict):
+    fix_concept_dict = {}
+    prior_list = ['precise ingredient synonyms', 'brand name synonyms', 'ingredient synonyms', 'precise ingredient',
+                  'brand name', 'name']
+    # For each concept we create a dictionary of priorities and then rebuild the concept using the dictionary
+    # without duplication between aliases groups
+    for cid, v in concept_dict.items():
+        # Creating a dictionary where the key is concept name and the value is the alias group the name belongs to
+        prior_dict = {}
+        for al in prior_list:
+            if al == 'name':
+                prior_dict[v[al]] = al
+            else:
+                for n in v[al]:
+                    prior_dict[n] = al
+        # Turning the dictionary we created into a concept without duplication
+        fix_concept = {'name': '', 'ingredient synonyms': [], 'precise ingredient': [],
+                       'brand name': [], 'precise ingredient synonyms': [], 'brand name synonyms': []}
+        for n, al in prior_dict.items():
+            if al == 'name':
+                fix_concept[al] = n
+            else:
+                fix_concept[al].append(n)
+        fix_concept_dict[cid] = fix_concept
+    return fix_concept_dict
+
+
+# A function that returns a set of the concepts id that don't have a line containing RXNORM and also have lines whose TTY is IN\BN\PIN
+def get_con_no_rxnorm():
+    with open("rxnorm/RXNCONSO.RRF") as con:
+        row_iter = csv.DictReader(con, delimiter='|', fieldnames=["RXCUI", "LAT", "TS", "LUI", "STT", "SUI", "ISPREF",
+                                                                  "RXAUI", "SAUI", "SCUI", "SDUI", "SAB", "TTY", "CODE",
+                                                                  "STR", "SRL", "SUPPRESS", "CVF"])
+
+        con_row = [dict(x) for x in list(row_iter)]
+        rxnorm_con = set()
+        con_id_bn_in_pin = set()
+
+        for row in con_row:
+            if row['TTY'] in ('BN', 'IN', 'PIN') and row['SAB'] == 'RXNORM':
+                rxnorm_con.add(row['RXCUI'])
+
+        for row in con_row:
+            if row['TTY'] in ('BN', 'IN', 'PIN') and row['RXCUI'] not in rxnorm_con:
+                con_id_bn_in_pin.add(row['RXCUI'])
+
+    return con_id_bn_in_pin
```

Comment:
General comments to this file:
- Looks good! i left some minor comments.
- (dont change, but)
  - I would split the `rxnorm_extract` function to a few more functions for readability and maintainability.
  - I would try to make it more optimized (as we run on all the lines a few times)
  - I would try to simplify the code a its quite complex (maybe applying the two lines before this one would already make it more simple)